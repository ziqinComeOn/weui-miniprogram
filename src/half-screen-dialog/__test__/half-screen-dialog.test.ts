import path from 'path'
import simulate from 'miniprogram-simulate'

describe('grids', () => {
    const halfScreenDialog = simulate.load(
        path.resolve(__dirname, '../half-screen-dialog'),
        'mp-half-screen-dialog'
    )

    test('basic', async () => {
        const onButtonTap = jest.fn()
        const onClose = jest.fn()

        const wrapper = simulate.render(
            simulate.load({
                usingComponents: {
                    'mp-half-screen-dialog': halfScreenDialog
                },
                template: `
                    <mp-half-screen-dialog id="dialog" show="{{show}}" buttons="{{buttons}}" bindbuttontap="onButtonTap" bindclose="onClose">
                        <view slot="title">测试标题A</view>
                        <view slot="desc">这是自定义描述区</view>
                    </mp-half-screen-dialog>
                `,
                data: {
                    show: false,
                    buttons: [
                        {
                            text: '按钮A',
                            type: 'default'
                        },
                        {
                            text: '按钮B',
                            type: 'primary'
                        }
                    ]
                },
                methods: {
                    onButtonTap,
                    onClose
                }
            })
        )
        wrapper.attach(document.createElement('parent-wrapper'))
        const comp = wrapper.querySelector('#dialog')
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()

        comp.setData({ show: true })
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()

        const buttons = comp.querySelectorAll('.weui-btn')
        buttons[1].dispatchEvent('tap')
        await simulate.sleep(0)
        expect(onButtonTap).toBeCalledTimes(1)
        expect(onButtonTap.mock.calls[0][0].detail).toStrictEqual({
            index: 1,
            item: {
                text: '按钮B',
                type: 'primary'
            }
        })

        const mask = comp.querySelector('.weui-mask')
        mask.dispatchEvent('tap')
        await simulate.sleep(0)
        expect(onClose).toBeCalledTimes(1)
    })
})
