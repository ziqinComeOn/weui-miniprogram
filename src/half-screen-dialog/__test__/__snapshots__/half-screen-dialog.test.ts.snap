// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grids basic 1`] = `
<mp-half-screen-dialog
  id="dialog"
  bind:buttontap="onButtonTap"
  bind:close="onClose"
>
  <wx-view
    class="weui-hidden"
  >
    <wx-view
      class="weui-mask init"
      data-type="tap"
      bind:tap="close"
      catch:touchmove="onMaskMouseMove"
    />
    <wx-view
      class="weui-half-screen-dialog "
    >
      <wx-view
        class="weui-half-screen-dialog__hd"
      >
        <wx-view
          class="weui-half-screen-dialog__hd__side"
          data-type="close"
          bind:tap="close"
        >
          <wx-view
            class="weui-icon-btn weui-icon-btn_close"
          >
            关闭
          </wx-view>
        </wx-view>
        <wx-view
          class="weui-half-screen-dialog__hd__main"
        >
          <wx-view
            class="weui-half-screen-dialog__title"
          >
            <wx-view
              slot="title"
            >
              测试标题A
            </wx-view>
          </wx-view>
        </wx-view>
        <wx-view
          class="weui-half-screen-dialog__hd__side"
        >
          <wx-view
            class="weui-icon-btn weui-icon-btn_more"
          >
            更多
          </wx-view>
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-half-screen-dialog__bd"
      >
        <wx-view
          slot="desc"
        >
          这是自定义描述区
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-half-screen-dialog__ft"
      >
        <wx-button
          class="weui-btn "
          data-index="{{0}}"
          type="default"
          bind:tap="buttonTap"
        >
          按钮A
        </wx-button>
        <wx-button
          class="weui-btn "
          data-index="{{1}}"
          type="primary"
          bind:tap="buttonTap"
        >
          按钮B
        </wx-button>
      </wx-view>
    </wx-view>
  </wx-view>
</mp-half-screen-dialog>
`;

exports[`grids basic 2`] = `
<mp-half-screen-dialog
  id="dialog"
  bind:buttontap="onButtonTap"
  bind:close="onClose"
>
  <wx-view
    class="weui-show"
  >
    <wx-view
      class="weui-mask init"
      data-type="tap"
      bind:tap="close"
      catch:touchmove="onMaskMouseMove"
    />
    <wx-view
      class="weui-half-screen-dialog "
    >
      <wx-view
        class="weui-half-screen-dialog__hd"
      >
        <wx-view
          class="weui-half-screen-dialog__hd__side"
          data-type="close"
          bind:tap="close"
        >
          <wx-view
            class="weui-icon-btn weui-icon-btn_close"
          >
            关闭
          </wx-view>
        </wx-view>
        <wx-view
          class="weui-half-screen-dialog__hd__main"
        >
          <wx-view
            class="weui-half-screen-dialog__title"
          >
            <wx-view
              slot="title"
            >
              测试标题A
            </wx-view>
          </wx-view>
        </wx-view>
        <wx-view
          class="weui-half-screen-dialog__hd__side"
        >
          <wx-view
            class="weui-icon-btn weui-icon-btn_more"
          >
            更多
          </wx-view>
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-half-screen-dialog__bd"
      >
        <wx-view
          slot="desc"
        >
          这是自定义描述区
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-half-screen-dialog__ft"
      >
        <wx-button
          class="weui-btn "
          data-index="{{0}}"
          type="default"
          bind:tap="buttonTap"
        >
          按钮A
        </wx-button>
        <wx-button
          class="weui-btn "
          data-index="{{1}}"
          type="primary"
          bind:tap="buttonTap"
        >
          按钮B
        </wx-button>
      </wx-view>
    </wx-view>
  </wx-view>
</mp-half-screen-dialog>
`;
